routes/user.js

```diff
@@ -0,0 +1,182 @@
+import mongoose from 'mongoose' ;
+
+
+// Schema For the User looking / Posting Opprtunities 
+let userSchema=  new mongoose.Schema(
+
+{
+    username:{
+    type : String ,
+    required: true ,
+    default : 'NA' ,
+    minLength: 5,
+    maxLength: 100,
+    } ,
+
+    password :{
+
+        type : String ,
+        required: true ,
+        default :'NA',
+        minLength: 8,
+        maxLength: 100,
+    } ,
+
+    gender :{
+ 
+        type: String ,
+        required: true ,
+        default: 'Others' 
+    } ,
+ 
+    dateofBirth : {
+   type : Date ,
+   required :true ,
+   default : new Date().getDate 
+
+    } ,
+    // to confirm wether the person is in college or already working
+
+    isWorking: Boolean , 
+    
+
+// Personal/Description similar to linkedin Summary 
+
+    briefDescription: {
+
+        type :String ,
+        required :true ,
+        maxLength: 100 
+    } ,
+
+    collegeName:{
+        type: String ,
+        required: true ,
+        default: 'NA' ,
+        
+    } ,
+
+     // Get both state and city of the user's College/university
+    collegeState:{          
+
+        type: String ,
+        required: true ,
+        default :' NA'
+    } ,
+    collegeCity:{          
+
+        type: String ,
+        required: true ,
+        default :' NA'
+    } ,
+
+   yearOfGraduation: {
+       type: Date ,
+       default : new Date().getFullYear(),
+       required: true 
```

Comment:
not required, We will keep only the bare required things as required so that users aren't irritated while completing their profiles.